nautobot/extras/jobs.py

```diff
@@ -104,25 +105,185 @@ class Meta:
         return str(self.name)
 
     # See https://github.com/PyCQA/pylint-django/issues/240 for why we have a pylint disable on each classproperty below
 
+    # TODO(jathan): Could be interesting for custom stuff when the Job is
+    # enabled in the database and then therefore registered in Celery
+    @classmethod
+    def on_bound(cls, app):
+        """Called when the task is bound to an app.
+
+        Note:
+            This class method can be defined to do additional actions when
+            the task class is bound to an app.
+        """
+
+    # TODO(jathan): Could be interesting for showing the Job's class path as the
+    # shadow name vs. the Celery task_name?
+    def shadow_name(self, args, kwargs, options):
+        """Override for custom task name in worker logs/monitoring.
+
+        Example:
+            .. code-block:: python
+
+                from celery.utils.imports import qualname
+
+                def shadow_name(task, args, kwargs, options):
+                    return qualname(args[0])
+
+                @app.task(shadow_name=shadow_name, serializer='pickle')
+                def apply_function_async(fun, *args, **kwargs):
+                    return fun(*args, **kwargs)
+
+        Arguments:
+            args (Tuple): Task positional arguments.
+            kwargs (Dict): Task keyword arguments.
+            options (Dict): Task execution options.
+        """
+
+    def before_start(self, task_id, args, kwargs):
+        """Handler called before the task starts.
+
+        .. versionadded:: 5.2
+
+        Arguments:
+            task_id (str): Unique id of the task to execute.
+            args (Tuple): Original arguments for the task to execute.
+            kwargs (Dict): Original keyword arguments for the task to execute.
+
+        Returns:
+            None: The return value of this handler is ignored.
+        """
+        self.active_test = "initialization"
+        self._job_result = None
+
+        try:
+            self.get_job_result()
+        except TypeError as err:
+            raise RunJobTaskFailed(f"Unable to serialize data for job {task_id}") from err
+        except Exception as err:
+            raise RunJobTaskFailed(f"Unexpected failure in job {self.name}") from err
+
+        job_model = self.get_job_model()
+        if not job_model.enabled:
+            self.log_failure(
+                message=f"Job {job_model} is not enabled to be run!",
+                obj=job_model,
+            )
+
+        soft_time_limit = job_model.soft_time_limit or settings.CELERY_TASK_SOFT_TIME_LIMIT
+        time_limit = job_model.time_limit or settings.CELERY_TASK_TIME_LIMIT
+        if time_limit <= soft_time_limit:
+            self.log_warning(
+                f"The hard time limit of {time_limit} seconds is less than "
+                f"or equal to the soft time limit of {soft_time_limit} seconds. "
+                f"This job will fail silently after {time_limit} seconds.",
+            )
```

Comment:
I feel like this is something that should be done at `JobResult.enqueue_job()` time and not at runtime? Or perhaps even have it as a part of `JobResult` model validation?

Reply:
Wouldn't be a bad idea to check at model validation time  (prevent explicitly configuring `job_model.time_limit <= job_model.soft_time_limit`) but I think it's appropriate to also warn at job execution time in the case where the settings are misconfigured.

(Actually, maybe that should be a Django check we should add too?)